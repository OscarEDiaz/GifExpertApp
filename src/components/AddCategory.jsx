import { useState } from "react"

export const AddCategory = ({onNewCategory}) => {
    const [inputValue, setInputValue] = useState("");

    const onInputChange = ({ target: { value } }) => {
        setInputValue(value);
    }

    const onSubmit = (e) => {
        e.preventDefault();
        const stringLength = inputValue.trim().length;
        
        if (stringLength <= 1) return;

        onNewCategory(inputValue);
        setInputValue('');
    }

    return (
        <form onSubmit={onSubmit} aria-label="form">
            <input 
                type="text"
                placeholder="Search gifs"
                value={inputValue}
                onChange={onInputChange}
            />
        </form>
    )
}
