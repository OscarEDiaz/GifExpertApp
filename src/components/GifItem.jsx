import PropTypes from 'prop-types';

export const GifItem = ({title, url, id, onDeletePressed}) => {
    const deletePressed = () => {
        onDeletePressed(id);
    }

    return (
        <div className="card">
            <button onClick={deletePressed} className='delete-btn'>Delete</button>
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    )
}

GifItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
    id: PropTypes.string.isRequired
}