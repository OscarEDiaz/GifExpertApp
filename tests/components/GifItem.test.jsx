import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Pruebas en componente GifItem', () => {
    const title = "TEST";
    const url = "https://hola.com/test.jpg"

    test('Debe hacer match con el snapshot', () => {
        const { container } = render(<GifItem title={title} url={url} />)
        expect(container).toMatchSnapshot();
    })

    test('Debe de mostrar la imagen con el URL y ALT indicado', () => {
        render(<GifItem title={title} url={url} />);
        
        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    })

    test('Debe de renderizar el tÃ­tulo', () => {
        render(<GifItem title={title} url={url} />);

        expect(screen.getByText(title)).toBeTruthy();
    })
})