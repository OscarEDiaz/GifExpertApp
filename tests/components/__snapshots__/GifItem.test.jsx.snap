// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente GifItem Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="TEST"
      src="https://hola.com/test.jpg"
    />
    <p>
      TEST
    </p>
  </div>
</div>
`;
